events/bus.go

```diff
@@ -10,7 +10,7 @@ import (
 
 // EventBus manages the state of and transmits messages to all its Subscribers
 type EventBus struct {
-	registry map[Subscriber]bool
+	registry map[*Subscriber]bool
```

Comment:
One of the things I'm not sure about was this change. I had to make it in order for the compiler to pass once I started passing `EventPublisher` through other changes, but, admittedly, I don't fully understand the repercussions here.